components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -192,17 +192,31 @@ handler.on(WalletActions.removeFavoriteApp.getType(), async (store: Store, appIt
   await refreshWalletInfo(store)
 })
 
+handler.on(WalletActions.chainChangedEvent.getType(), async (store: Store, payload: ChainChangedEventPayloadType) => {
+  const keyringService = getAPIProxy().keyringService
+  const state = getWalletState(store)
+  const { accounts } = state
+  const selectedAccountAddress = await keyringService.getSelectedAccount(payload.coin)
+  const selectedAccount = accounts.find((account) => account.address === selectedAccountAddress.address) ?? accounts[0]
+  store.dispatch(WalletActions.setSelectedAccount(selectedAccount))
+  store.dispatch(WalletActions.setSelectedCoin(payload.coin))
```

Comment:
Why do we need to remember the selected coin if we already have this information from the `coin` field of `AccountInfo`?

Reply:
This is needed since each `coinType` has a `defaultNetwork` and `defaultAccount`.
With out keeping track of the selected `coinType`, we have no way to keep track in the `Panel` and `Buy/Send/Swap` widgets of what a user has selected.